### PR TITLE
Deduplication automation

### PR DESCRIPTION
**Pull request by mmollick**
Hey saw this project on the Cleveland Tech slack and thought I could improve the workflow a bit.

1. This adds a `--download` flag to the `import.py`, this will automatically download the Google Sheet. (Sort of a duplicate of #8 )
2. Add people deduplication based on app usernames
3. Adds support for known place aliases to automate deduplication based around common spelling mistakes and aliases. 

For example with:
```json
{
    "name": "Frank and Tony\u2019s",
    "website": "",
    "note": "",
    "workers": [
      {
        "name": "Brittany Berwald",
        "paypal": "BrittanyBerwald",
        "venmo": "brittyaileen",
        "cashapp": "baileen1220",
        "last4": "7404"
      }
    ]
  },
  {
    "name": "Frank and Tony\u2019s Place",
    "website": "",
    "note": "",
    "workers": [
      {
        "name": "Kiera Labarbera",
        "paypal": "",
        "venmo": "Kiera-Labarbera",
        "cashapp": "",
        "last4": "8493"
      }
    ],
    "alias": [
      "Frank and Tony\u2019s",
      "Frank and Tony\u2019s place"
    ]
  },
```
Rerunning `import.py` will condense these into:
```json
  {
    "name": "Frank and Tony\u2019s Place",
    "website": "",
    "note": "",
    "workers": [
     {
        "name": "Brittany Berwald",
        "paypal": "BrittanyBerwald",
        "venmo": "brittyaileen",
        "cashapp": "baileen1220",
        "last4": "7404"
      },
      {
        "name": "Kiera Labarbera",
        "paypal": "",
        "venmo": "Kiera-Labarbera",
        "cashapp": "",
        "last4": "8493"
      }
    ],
    "alias": [
      "Frank and Tony\u2019s",
      "Frank and Tony\u2019s place"
    ]
  },
```

### CODE REVIEW

File: import.py [hunk 1]
```diff
@@ -80,7 +87,12 @@ def parse_username(username: str):
 for person in responses:
     person_exists = False
     for worker in existing_workers:
-        if person["name"].lower() == worker["name"].lower():
+
+        # prevent duplicate entries with same name or app username
+        if person["name"].lower() == worker["name"].lower() \
+        or person["paypal"].lower() == worker["paypal"].lower() \
+        or person["venmo"].lower() == worker["venmo"].lower() \
+        or person["cashapp"].lower() == worker["cashapp"].lower():
```
Comment (vsalvino) [inline import.py:95]: Most often these will be blank, so it could trigger a false positive if "" == "" (only name is a required field)

**Comment by vsalvino** [comment]
This is a great idea. I'm wondering if we can help automate the alias adding process, by maybe comparing the names after stripping spaces, punctuation, etc - if the stripped versions match, then add an alias. IDK how effective that would be?

File: import.py (hunk 1 above)
Comment (mmollick) [reply import.py:95]: Good catch, we can check that the person contains a non empty value first.

**Comment by mmollick** [comment]
I'm not sure that would be effective. With the example in the original PR that method wouldn't catch the change. I think we would need to compare the words within the place name, look for ordering and likeness. We may even need to calculate the Levenshtein distance to effectively detect mispelled words. Even then we would still need to manually review the aliases it detects, since we're likely to get false positives.

**Pull request merged**